services/wallet/api.go

```diff
@@ -742,11 +755,185 @@ func (api *API) CreateMultiTransaction(ctx context.Context, multiTransactionComm
+	if inputParams.AmountIn != nil {
+		sd.FromAmount = inputParams.AmountIn.String()
+	}
+	if inputParams.AmountOut != nil {
+		sd.ToAmount = inputParams.AmountOut.String()
+	}
+	sd.OwnerTokenBeingSent = inputParams.TokenIDIsOwnerToken
+	sd.Username = inputParams.Username
+	sd.PublicKey = inputParams.PublicKey
+	if inputParams.PackID != nil {
+		sd.PackID = inputParams.PackID.String()
+	}
+}
+
+func (api *API) BuildTransactionsFromRoute(ctx context.Context, buildInputParams *requests.RouterBuildTransactionsParams) {
+	log.Debug("[WalletAPI::BuildTransactionsFromRoute] builds transactions from the generated best route", "uuid", buildInputParams.Uuid)
+
+	go func() {
+		api.router.StopSuggestedRoutesAsyncCalculation()
+
+		var err error
+		response := &responses.RouterTransactionsForSigning{
+			SendDetails: &responses.SendDetails{
+				Uuid: buildInputParams.Uuid,
+			},
+		}
+
+		defer func() {
+			if err != nil {
+				api.s.transactionManager.ClearLocalRouterTransactionsData()
+				err = statusErrors.CreateErrorResponseFromError(err)
+				response.SendDetails.ErrorResponse = err.(*statusErrors.ErrorResponse)
+			}
+			signal.SendWalletEvent(signal.SignRouterTransactions, response)
+		}()
+
+		route, routeInputParams := api.router.GetBestRouteAndAssociatedInputParams()
+		if routeInputParams.Uuid != buildInputParams.Uuid {
+			// should never be here
+			err = ErrCannotResolveRouteId
+			return
+		}
+
+		updateFields(response.SendDetails, routeInputParams)
+
+		// notify clinet that sending transactions started (has 3 steps, building txs, signing txs, sending txs)
+		signal.SendWalletEvent(signal.RouterSendingTransactionsStarted, response.SendDetails)
+
+		response.SigningDetails, err = api.s.transactionManager.BuildTransactionsFromRoute(
+			route,
+			routeInputParams.AddrFrom,
+			routeInputParams.AddrTo,
+			api.router.GetPathProcessors(),
+			routeInputParams.Username,
+			routeInputParams.PublicKey,
+			routeInputParams.PackID.ToInt(),
+			buildInputParams.SlippagePercentage)
+	}()
+}
+
+// @deprecated `ProceedWithTransactionsSignatures`
+//
+// The flow that should be used instead:
+// - call `BuildTransactionsFromRoute`
+// - wait for the `wallet.router.sign-transactions` signal
+// - sign received hashes using `SignMessage` call or sign on keycard
+// - call `SendRouterTransactionsWithSignatures` with the signatures of signed hashes from the previous step
+//
+// TODO: remove this struct once mobile switches to the new approach
 func (api *API) ProceedWithTransactionsSignatures(ctx context.Context, signatures map[string]transfer.SignatureDetails) (*transfer.MultiTransactionCommandResult, error) {
 	log.Debug("[WalletAPI:: ProceedWithTransactionsSignatures] sign with signatures and send multi transaction")
 	return api.s.transactionManager.ProceedWithTransactionsSignatures(ctx, signatures)
 }
 
+func (api *API) SendRouterTransactionsWithSignatures(ctx context.Context, sendInputParams *requests.RouterSendTransactionsParams) {
```

Comment:
Why do API endpoints end up so long? 🤔 
I thought that in most cases they should be a simple wrapper around some Go method

Reply:
Cause moving all that out of API requires many other changes. I agree, we should do that through another PR.